Dyad ranker tests

JonasHanselle opened the pull request:
This pull request contains various changes. Among them:
* Loss/accuracy/correlation functions and utilities for them
* Test resources from Dirk Schäfers genetic algorithm traveling salesman problem dataset and the corresponding unit test
* Test resources from metamining dyad rankings from performance data randomly sampled from ML-Plans Weka searchspace and the corresponding unit test
* Prototypical serialization and deserialization for dyad ranking datasets and a corresponding unit test
* Minibatch learning for PLNet dyad ranker
* Early stopping retrain functionality for PLNet dyad ranker
* Standard scaler for dyad ranking datasets

Commit:
Add tests for dyad ranker & fix dyad ranker so that tests work

Commit:
Merge branch 'DyadRankerTests' of
https://github.com/AILibs-DyRaCo/AILibs.git into DyadRankerTests

Conflicts:
	JAICore/jaicore-ml/test/jaicore/ml/dyadranking/general/SimpleDyadDatasetDyadRankerTester.java

mirkojuergens committed:
adds seeds

Commit:
Add more loss functions

JonasHanselle committed:
Adds javadoc to the DyadRankingInstanceSupplier

JonasHanselle committed:
Adds javadoc to the DyadRankingInstanceSupplier

JonasHanselle committed:
Start working on dyad ranker test based on Dirk Schaefers GA TSP dataset

JonasHanselle committed:
Work on DyadRankerGATSPTest, add constructors to DyadRankingDataset, change equals method of Dyad such that it compares the instance and the alternative instead of object references

JonasHanselle committed:
Merge branch 'dev' into DyadRankerTests

JonasHanselle committed:
Minor changes

JonasHanselle committed:
Merge dev into DyadRankerTester

JonasHanselle committed:
Small adjustment

JonasHanselle committed:
Small fix

mibraun committed:
Implemented Kendall's tau for general dyad rankings.

JonasHanselle committed:
Adds stantandardization to PLNetDyadRanker

JonasHanselle committed:
Merge branch 'DyadRankerTests' of https://github.com/AILibs-DyRaCo/AILibs into DyadRankerTests

JonasHanselle committed:
small fix regarding standardization

JonasHanselle committed:
Standardize instance features in GATSPTest

JonasHanselle committed:
Fix test

JonasHanselle committed:
Adds the REAL orderings of the dyads and adjusts the DyadRankerGATSPTest to work with these

JonasHanselle committed:
Merge dev into DyadRankerTests

JonasHanselle committed:
Add loss utility and DyadDatasetGenerator

mibraun committed:
Cleaned up various log calls and commented out lines left over from debugging.

JonasHanselle committed:
Train and test data in DyadDatasetGenerator are now distinct

JonasHanselle committed:
Some cleanup and the DyadRankerGATSPTest now works with both DyadRankers implemented so far

JonasHanselle committed:
Adds a very simple serialization for DyadRankingDatasets

JonasHanselle committed:
Start work on DyadRankerMetaminingTest

mibraun committed:
Implemented option for plnet to retrain after early stopping.
Made shuffling of the data set before training of plnet use the fixed seed from the configuration.

mibraun committed:
- Option to use mini batches added to plNet.
- Removed shuffling of the training data in the plNet training method as it constituted a likely unwanted (and undocumented) side effect on the training data outside of the class.

mibraun committed:
NDCG score added

JonasHanselle committed:
Adds the DyadDatasetGenerator that was used for first evaluations of the PLNet

JonasHanselle committed:
Limits the number of threads for ND4L backend to 4

JonasHanselle committed:
Adds even more calls to limit the use of threads, apparently it does not work though

JonasHanselle committed:
Minor cleanup

JonasHanselle committed:
Removed TSP instances and other data that was not needed

JonasHanselle committed:
Work on equals methods in DyadRankingInstance and SparseDyadRankingInstance, serialization and deserialization of DyadRankingDatasets incl. unit test

mibraun committed:
Adjustments to NDCG

mibraun committed:
Cleaned up commented-out lines in plnet

JonasHanselle committed:
Removed DyadDatasetGenerator from repo

JonasHanselle committed:
Merge branch 'DyadRankerTests' of https://github.com/AILibs-DyRaCo/AILibs into DyadRankerTests

mibraun committed:
Removed unneeded shuffleData parameter.

JonasHanselle committed:
Work on standard scaler, metamining unit test

JonasHanselle committed:
Merge branch 'DyadRankerTests' of https://github.com/AILibs-DyRaCo/AILibs into DyadRankerTests

JonasHanselle committed:
Add javadoc to the DyadStandardScaler, clean up unit tests, add config to unit tests to make them run faster

JonasHanselle committed:
Adds doc to the DyadRankingDatasetSerializationTest

JonasHanselle committed:
Clean up tests, DyadRankerGATSPTest now uses the DyadStandardScaler instead standardizing the dataset manually

JonasHanselle committed:
Minor doc for DyadRankingDataset serialization and deserialization

JonasHanselle committed:
Adds doc to DyadRankerMetaminingTest

JonasHanselle committed:
Trim dyad rankings before standardizing them in DyadRankerGATSPTest to avoid data snooping

JonasHanselle committed:
Adds methods for printing STD and Mean in DyadStandardScaler

mibraun commented on JAICore/jaicore-ml/src/jaicore/ml/dyadranking/loss/DyadRankingLossFunction.java:
```diff
@@ -0,0 +1,25 @@
+ * @author Helena Graf
+ *
+ */
+public interface DyadRankingLossFunction {
```
We might want to rename this, as our "losses" aren't technically losses but something like scores, i.e. higher is better.

mibraun committed:
Minor typos fixed.
Implemented some unimplemented DyadRankingDataset utility methods (or made them throw UnsupportedOperationExceptions)

mibraun reviewed (approved):
Looks fine. We might want to rename "loss" to "score" at some point.

mirkojuergens committed:
Adds standford wrapper, precomputes bilin ft; unifies the implementation with the reference implementation

mirkojuergens committed:
Deletes DyadRankingDatasetGenerator

 as this is part of the experiment project
